app/routes/companyInterest/components/CompanyInterestPage.tsx

```diff
@@ -348,8 +358,10 @@ const CompanyInterestPage = (props: Props) => {
 
   const onSubmit = (data) => {
     const { company } = data;
-    const companyId = company['value'] ? Number(company['value']) : null;
-    const companyName = companyId === null ? company['label'] : '';
+    const nameOnly = company['__isNew__'] || !company.value;
+    const companyId = nameOnly ? null : company['value'];
+    const companyName = nameOnly ? company['label'] : '';
```

Comment:
```suggestion
    const companyId = nameOnly ? null : company.value;
    const companyName = nameOnly ? company.label : '';
```
Any specific reason for using squared brackets?

Reply:
not really, just didn't change it